### PR TITLE
Traefik generator support

### LINKED ISSUE #134: Traefik Proxy support
**Is your feature request related to a problem? Please describe.**
I want to expose my REST API with Traefik - but creating and maintaining that configuration is a hassle...

**Describe the solution you'd like**
I would like kusk to be able to generate Traefik configurations in line with https://doc.traefik.io/traefik/v2.3/routing/providers/kubernetes-crd/#custom-resource-definition-crd

## TODO
- [x] Detection in Cluster
- [x] Add generator
- [x] Add Docs
- [x] Add Wizard flow

### PR DESCRIPTION
**Pull request by Tarick**
This PR adds an initial support for Traefik IngressRoute generation

## Changes

* generation of Traefik IngressRoute and Middleware using CRD objects  and marshalling to Yaml
* added related tests
* wizard detection of intsalled Traefik based on available CRD API in  the cluster

## Fixes
resolves #134 

## Checklist

- [x] tested locally
- [x] added new dependencies
- [x] updated the docs
- [x] added a test

### CODE REVIEW

**Review by kylehodgetts — CHANGES_REQUESTED** [review]
Good initial PR!

File: cluster/cluster_test.go [hunk 1]
```diff
@@ -242,3 +242,48 @@ func TestClient_DetectLinkerd(t *testing.T) {
 		})
 	}
 }
+
+func TestClient_DetectTraefik(t *testing.T) {
+	// Test calling API for Traefik CRD
+	t.Run("Traefik CRD API is installed", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Tarick) [reply cluster/cluster_test.go:248]: This one is tricky - Fake client doesn't return "Not Found" error properly as a real client, so I have to ignore errors in "No Traefik" test and this makes it different from all others.

File: wizard/wizard.go [hunk 2]
```diff
@@ -122,10 +131,13 @@ func flowWithCluster(args *flow.Args, kubeConfigPath string) (string, error) {
 		return "", fmt.Errorf("failed to list namespaces: %w", err)
 	}
 
+	args.TargetNamespace = targetServiceNamespace
+
 	args.TargetService = args.Prompt.SelectOneOf("Choose your service", targetServiceSuggestions, true)
 
 	args.Service = args.Prompt.SelectOneOf("Choose a service you want Kusk generate manifests for", servicesToSuggest, false)
 
+
```
[the remark this answers is not in the document]
Comment (Tarick) [reply wizard/wizard.go]: Thanks, fixed!

File: cluster/cluster_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kylehodgetts) [reply cluster/cluster_test.go:248]: Feel free to resolve this conversation :) It's a nice to have

**Review by kylehodgetts — APPROVED** [review]

**Review by dobegor — APPROVED** [review]

**Pull request merged**